File: src/main/java/ru/trukhmanov/twochairsbackend/entity/DeckQuestionId.java
package ru.trukhmanov.twochairsbackend.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import lombok.*;

import java.io.Serializable;

@Embeddable
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
public class DeckQuestionId implements Serializable {

    @Column(name="deck_id")
    private Long deckId;

    @Column(name="question_id")
    private Long questionId;
}

File: src/main/kotlin/ru/trukhmanov/twochairsbackend/entity/DeckQuestionId.kt
package ru.trukhmanov.twochairsbackend.entity

import jakarta.persistence.Column
import jakarta.persistence.Embeddable
import java.io.Serializable

@Embeddable
data class DeckQuestionId(
    @Column(name = "deck_id")
    var deckId: Long? = null,

    @Column(name = "question_id")
    var questionId: Long? = null
) : Serializable
